things/service.go

```diff
@@ -147,6 +147,22 @@ type thingsService struct {
 	ulidProvider mainflux.IDProvider
 }
 
+// createUID: take the external id if supplied, otherwise use default provider
+// validate the external id, if validation fails, generate an id using the
+// default provider.
+func (ts *thingsService) createUID(extid string) (id string, err error) {
```

Comment:
Maybe better `provideUUID` and `externalID` or `extID`

Reply:
changed to extID. Thank You.